src/appleseed/foundation/math/dual.h

```diff
@@ -102,6 +104,13 @@ inline const T& Dual<T>::get_value() const
 }
 
 template <typename T>
+inline void Dual<T>::set_value(const T& value)
+{
+    m_value = value;
+    m_has_derivatives = false;
```

Comment:
It is expected that you call set_value() before set_derivatives().
But checking the code, we don't need setters at all and then there's no 
possibility of doing the wrong thing anymore.